src/analysis/processing/qgsalgorithmextractspecificvertices.h

```diff
@@ -0,0 +1,56 @@
+/***************************************************************************
+                         qgsalgorithmextractspecificvertices.h
+                         -------------------------
+    begin                : November 2019
+    copyright            : (C) 2019 by Alexander Bruy
+    email                : alexander dot bruy at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSALGORITHMEXTRACTSPECIFICVERTICES_H
+#define QGSALGORITHMEXTRACTSPECIFICVERTICES_H
+
+#define SIP_NO_FILE
+
+#include "qgis_sip.h"
+#include "qgsprocessingalgorithm.h"
+#include "qgsapplication.h"
+
+///@cond PRIVATE
+
+/**
+ * Native extract specific vertices algorithm.
+ */
+class QgsExtractSpecificVerticesAlgorithm : public QgsProcessingAlgorithm
```

Comment:
Hm, I wonder why we never made this one a feature based algorithm. Probably just an oversight. (We can address this later anyway)

Reply:
I was thinking about making it feature based but then looked at the "Extract vertices" C++ algorithm and found that itis also layer based. So I assumed there was some reasons to make it layer based and ported it as layer based too.

Will address this later if you don't object.

Reply:
I'm guessing it was just an oversight -- I can't see any reason this couldn't/shouldn't be feature based.